### PR TITLE
Tidy python bindings in the endian_swapper example

### PR DESCRIPTION
**Pull request by eric-wieser**
See individual commit messages.

It's possible but unlikely that this solves @ktbarrett's bug report from gitter, https://gitter.im/cocotb/Lobby?at=5e977cf563e7b73a5fda24af

### CODE REVIEW

File: examples/endian_swapper/cosim/io.c [hunk 1]
```diff
@@ -36,84 +36,78 @@
...
-    if (PyErr_Occurred())
-        PyErr_Print();
-
     Py_DECREF(rv);
-    Py_DECREF(call_args);
-
     return 0;
 }
 
-#if PY_MAJOR_VERSION >= 3
```
Comment (eric-wieser) [inline examples/endian_swapper/cosim/io.c:116]: xref #1289

**Comment by themperek** [comment]
Same error for me whit swig 2.0.10

**Comment by themperek** [comment]
Maybe we should run this test only in CI for now since documentation point to this as the first things to do and it is has many issues not related to what is important.

**Comment by eric-wieser** [comment]
Agreed, the big issue with this example right now is that it's the one we point users to first.

**Comment by imphil** [comment]
To keep this PR focused I moved the discussion beyond that to https://github.com/cocotb/cocotb/issues/1660.

**Pull request merged**